lib/Doctrine/ORM/AbstractQuery.php

```diff
@@ -88,7 +88,7 @@ abstract class AbstractQuery
     /**
      * The user-specified ResultSetMapping to use.
      */
-    protected ?ResultSetMapping $_resultSetMapping = null;
+    protected ResultSetMapping|null $_resultSetMapping = null;
```

Comment:
What's the rationale behind enforcing the usage of union types instead of nullable types?

Reply:
https://github.com/doctrine/coding-standard/pull/266#discussion_r900753185

Reply:
Personally, I believe arbitrary union types (e.g. `getValue(sting $key): int|string|false`) often reflect poor API design while nullable types are fine (e.g. `findUser(int $id): ?User`).

I think that enforcing a feature introduced to support poor design as a one-size-fits-all solution is not the right move.

Reply:
Union types are fine. I must disagree that union types often reflect poor API design. I think the premise is too simplified.

Reply:
We have two ways of expressing the same thing: a union of a type and `null`. Normalizing that is reasonable imho. Whether we use the old nullable syntax or the new union syntax does not matter that much to me, as long as we agree on a single way to do it.

And I don't follow that "poor API design" argument either.

Reply:
All right, so how do you suggest to proceed here?

Reply:
We do not forbid nullable types in the coding standard: https://github.com/doctrine/coding-standard/pull/266#discussion_r900749599.

Reply:
> We do not forbid nullable types in the coding standard

That means, we would allow both ways of expressing nullable types? Or would we let the fixer change `Foo|null` to `?Foo`?

Do you want to open a PR on the coding-standards repo that makes the necessary config adjustments? I think, this change should be discussed there.

Reply:
I think different topic is discussed here. The sniff is not about whether or not it is ok to design API with unions. It's not about API at all.

The sniff is about consistent usage of union types. If there are no unions, the sniff does not apply.

If there's union of `Type` and `null`, it should be written as union: `Type|null`. The archaic `?Type` was introduced because of absence of union types in the language. Since there's unions support now, there's no reason to use `?` because it's inconsistent. We already have `null` value and [will have `null` standalone type in 8.2](https://wiki.php.net/rfc/null-false-standalone-types).

Reply:
> That means, we would allow both ways of expressing nullable types? Or would we let the fixer change Foo|null to ?Foo?

If the latter is possible, I would prefer that. Otherwise, the former is fine.

> Do you want to open a PR on the coding-standards repo that makes the necessary config adjustments?

No.